File: bffs/mobile-bff/app/routers/auth_routes.py
from fastapi import APIRouter, HTTPException
import httpx

from app.clients.user_client import register_user, login_user
from app.schemas.mobile_schemas import RegisterRequest, LoginRequest

router = APIRouter(prefix="/api/mobile/auth", tags=["Mobile Auth"])


@router.post("/register")
async def register(request: RegisterRequest):
    try:
        return await register_user(request.model_dump(exclude_none=True))
    except httpx.HTTPStatusError as ex:
        raise HTTPException(
            status_code=ex.response.status_code,
            detail=ex.response.json()
        )


@router.post("/login")
async def login(request: LoginRequest):
    try:
        return await login_user(request.model_dump())
    except httpx.HTTPStatusError as ex:
        raise HTTPException(
            status_code=ex.response.status_code,
            detail=ex.response.json()
        )

File: bffs/web-bff/src/routes/authRoutes.js
const express = require("express");
const userClient = require("../clients/userClient");

const router = express.Router();

router.post("/register", async (req, res, next) => {
  try {
    const result = await userClient.register(req.body);
    return res.status(201).json(result);
  } catch (error) {
    return next(error);
  }
});

router.post("/login", async (req, res, next) => {
  try {
    const result = await userClient.login(req.body);
    return res.json(result);
  } catch (error) {
    return next(error);
  }
});

module.exports = router;